### PR TITLE
extract abstract explain statement & support postgresql explain statement

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #10368.

Changes proposed in this pull request:
- rename mysql describe to explain
- extract abstract explain statement
- support postgresql explain statement
- correct mysql explain statement

### CODE REVIEW

**Review by taojintianxia — APPROVED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ExplainStatementTestCase.java [hunk 1]
```diff
@@ -22,5 +22,5 @@
 /**
  * Describe statement test case.
  */
-public final class DescribeStatementTestCase extends SQLParserTestCase {
+public final class ExplainStatementTestCase extends SQLParserTestCase {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ExplainStatementTestCase.java:25]: We do not have tests to assert its `SQLStatement` field currently, which is possible to be some community issues for our volunteers.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dal/ExplainStatementContext.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        SQLStatement explainableStatement = sqlStatement.getStatement().orElse(null);
+        TableExtractor extractor = new TableExtractor();
+        if (explainableStatement instanceof SelectStatement) {
+            extractor.extractTablesFromSelect((SelectStatement) explainableStatement);
+        } else if (explainableStatement instanceof InsertStatement) {
+            extractor.extractTablesFromInsert((InsertStatement) explainableStatement);
+        } else if (explainableStatement instanceof UpdateStatement) {
+            extractor.extractTablesFromUpdate((UpdateStatement) explainableStatement);
+        } else if (explainableStatement instanceof DeleteStatement) {
+            extractor.extractTablesFromDelete((DeleteStatement) explainableStatement);
```
[the remark this answers is not in the document]
Comment (strongduanmu) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dal/ExplainStatementContext.java:62]: > extractTablesFromSQLStatement

@tristaZero This is a great suggestion. There are similar logic in other places. Extracting to TableExtractor will make the code clearer. 👍

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/handler/dal/ExplainStatementHandler.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     * Get simple table segment.
+     *
+     * @param explainStatement explain statement
+     * @return simple table segment
+     */
+    public static Optional<SimpleTableSegment> getSimpleTableSegment(final ExplainStatement explainStatement) {
+        if (explainStatement instanceof MySQLStatement) {
+            return ((MySQLExplainStatement) explainStatement).getTable();
+        }
+        return Optional.empty();
```
[the remark this answers is not in the document]
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/handler/dal/ExplainStatementHandler.java:46]: > So `PostgreSQLExplainStatement` is to-do one?

@tristaZero `ExplainStatementHandler` is used to deal with fields that are unique to different db dialects. For mysql and postgresql, `sqlStatement` is a common field, and the `table` field only exists in the mysql dialect, so only `MySQLStatement` is processed here.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLExplainStatement.java [hunk 4]
```diff
@@ -17,21 +17,29 @@
...
- * MySQL describe statement.
+ * MySQL explain statement.
  */
-@Getter
 @Setter
 @ToString
-public final class MySQLDescribeStatement extends AbstractSQLStatement implements DALStatement, MySQLStatement {
+public final class MySQLExplainStatement extends ExplainStatement implements MySQLStatement {
     
     private SimpleTableSegment table;
```
[the remark this answers is not in the document]
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLExplainStatement.java:35]: > This field will be removed `SQLStatement` later on, won't it?

@tristaZero In MySQL, explain also has the describe function. When explain is used in this scenario, a `table` field will be parsed, so this field is necessary. 

You can refer to the explain grammar:
```
{EXPLAIN | DESCRIBE | DESC}
    tbl_name [col_name | wild]

{EXPLAIN | DESCRIBE | DESC}
    [explain_type]
    {explainable_stmt | FOR CONNECTION connection_id}

{EXPLAIN | DESCRIBE | DESC} ANALYZE [FORMAT = TREE] select_statement

explain_type: {
    FORMAT = format_name
}

format_name: {
    TRADITIONAL
  | JSON
  | TREE
}

explainable_stmt: {
    SELECT statement
  | TABLE statement
  | DELETE statement
  | INSERT statement
  | REPLACE statement
  | UPDATE statement
}
```

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ExplainStatementTestCase.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ExplainStatementTestCase.java:25]: > We do not have tests to assert its `SQLStatement` field currently, which is possible to be some community issues for our volunteers.

@tristaZero Great suggestion, I will create some community tasks later. 👍

**Review by tristaZero — APPROVED** [review]

**Pull request merged**